### PR TITLE
Fix: env0_environment does not redeploy when revision is changed in w…

### LINKED ISSUE #906: `env0_environment` does not redeploy when revision is changed in `without_template_settings`
**Describe the bug**
When you change the `revision` field of an `env0_environment` resource - you will trigger a deployment. This is good. But when you have `without_template_settings`, you can't specify the `revision` directly on the `environment`, only nested within the `without_template_settings`, and when you change it - it **does not trigger a redeploy**.

**Expected behavior**
When changing the `revision` inside a `without_template_settings` block of an `env0_environment` resource, I expect the update to trigger a redeploy.

### PR DESCRIPTION
**Pull request by TomerHeber**
…ithout_template_settings

### Issue & Steps to Reproduce / Feature Request
fixes #906 

### Solution

1. Update shouldDeploy to test that use-cases.
2. Some other small changes.
3. Updated the acceptance tests.

### CODE REVIEW

File: env0/resource_environment.go [hunk 1]
```diff
@@ -732,6 +732,12 @@ func shouldUpdateTemplate(d *schema.ResourceData) bool {
 }
 
 func shouldDeploy(d *schema.ResourceData) bool {
+	if _, ok := d.GetOk("without_template_settings.0"); ok {
```
Comment (TomerHeber) [inline env0/resource_environment.go:735]: added the required use-case.

File: env0/resource_environment.go [hunk 2]
```diff
@@ -1126,8 +1132,7 @@ func getDeployPayload(d *schema.ResourceData, apiClient client.ApiClientInterfac
 	var err error
 
 	if isTemplateless(d) {
-		templateId, ok := d.GetOk("without_template_settings.0.id")
-		if ok {
+		if templateId, ok := d.GetOk("without_template_settings.0.id"); ok {
```
Comment (TomerHeber) [inline env0/resource_environment.go:1135]: no change... just one less line.

File: env0/resource_environment.go [hunk 3]
```diff
@@ -1139,6 +1144,10 @@ func getDeployPayload(d *schema.ResourceData, apiClient client.ApiClientInterfac
 	}
 
 	if isRedeploy {
+		if revision, ok := d.GetOk("without_template_settings.0.revision"); ok {
```
Comment (TomerHeber) [inline env0/resource_environment.go:1147]: set the revision in the deploy request only if it's a redeploy.

File: env0/resource_environment_test.go [hunk 4]
```diff
@@ -2541,6 +2541,13 @@ func TestUnitEnvironmentWithoutTemplateResource(t *testing.T) {
 
 				// Update
 				mock.EXPECT().TemplateUpdate(template.Id, templateUpdatePayload).Times(1).Return(updatedTemplate, nil),
+				mock.EXPECT().ConfigurationSetsAssignments("ENVIRONMENT", environment.Id).Times(1).Return(nil, nil),
+				mock.EXPECT().EnvironmentDeploy(environment.Id, client.DeployRequest{
```
Comment (TomerHeber) [inline env0/resource_environment_test.go:2545]: Now deploy is called as part of the test.

File: env0/resource_environment.go (hunk 2 above)
Comment (avnerenv0) [reply env0/resource_environment.go:1135]: this syntax is super strange ... do we use it a lot in this codebase / is it common in Go?

**Review by avnerenv0 — APPROVED** [review]
Thank you

File: env0/resource_environment.go (hunk 2 above)
Comment (TomerHeber) [reply env0/resource_environment.go:1135]: Yes.
This is common go.

**Pull request merged**